format.go

```diff
@@ -98,35 +101,124 @@ type protobufFormat struct{}
 
 // NewDecoder implements Format.
 func (protobufFormat) NewDecoder(r io.Reader) (FormatDecoder, error) {
-	return protobufWrapper{Reader: pbio.NewDelimitedReader(r, 1<<28)}, nil
+	return &protobufAdapter{reader: r}, nil
 }
 
 // NewEncoder implements Format.
 func (protobufFormat) NewEncoder(w io.Writer) (FormatEncoder, error) {
-	return protobufWrapper{Writer: pbio.NewDelimitedWriter(w)}, nil
+	return &protobufAdapter{writer: w}, nil
+}
+
+// protobufAdapter this is an adapter to switch between
+// protobufWrapper (official go protobuf impl) and
+// gogoProtobufWrapper (backcompat, deprecated - github.com/gogo/protobuf).
+//
+// Wrapper is instantiated on the first Encode or Decode call.
+// Either .writer or .reader must be set.
+type protobufAdapter struct {
```

Comment:
@dim this is the main change for back compat gogo Message handling

Reply:
I think this is really too complicated tbh, I'd rather just have:

```go
type protobufWrapper struct {
  w io.Writer
	pbw gio.Writer
 	enc *pbio.Encoder

  r io.Reader
  pbr	gio.Reader
  dec *pbio.Decoder
}
```

Reply:
Less switches though... But I don't think it's perf-critical code so will do what you suggest.